implementations/java-jsonschema-validator/bowtieJsonSchemaValidator.java

```diff
@@ -0,0 +1,215 @@
+      List.of("https://json-schema.org/draft/2020-12/schema",
+              "https://json-schema.org/draft/2019-09/schema",
+              "http://json-schema.org/draft-07/schema#",
+              "http://json-schema.org/draft-06/schema#",
+              "http://json-schema.org/draft-04/schema#");
+
+  private final ObjectMapper objectMapper = new ObjectMapper().configure(
+      DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+  private final JsonSchemaFactory schemaFactory =
+      JsonSchemaFactory.getInstance();
+  private final PrintStream output;
+
+  public static void main(String[] args) {
+    BufferedReader reader =
+        new BufferedReader(new InputStreamReader(System.in));
+    new BowtieJsonSchemaValidator(System.out).run(reader);
+  }
+
+  public BowtieJsonSchemaValidator(PrintStream output) { this.output = output; }
+
+  private void run(BufferedReader reader) {
+    reader.lines().forEach(this::handle);
+  }
+
+  private void handle(String data) {
+    try {
+      JsonNode node = objectMapper.readTree(data);
+      String cmd = node.get("cmd").asText();
+      switch (cmd) {
+        case "start" -> start(node);
+        case "dialect" -> dialect(node);
+        case "run" -> run(node);
+        case "stop" -> System.exit(0);
+        default -> throw new IllegalArgumentException(
+          "Unknown cmd [%s]".formatted(cmd)
+        );
+      }
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  private void start(JsonNode node) throws IOException {
+    StartRequest startRequest = objectMapper.treeToValue(
+      node,
+      StartRequest.class
+    );
+    if (startRequest.version() != 1) {
+      throw new IllegalArgumentException(
+        "Unsupported IHOP version [%d]".formatted(startRequest.version())
+      );
+    }
+
+    InputStream is = getClass().getResourceAsStream("META-INF/MANIFEST.MF");
+    var attributes = new Manifest(is).getMainAttributes();
+
+    StartResponse startResponse = new StartResponse(
+      1,
+      true,
+      new Implementation(
+        "java",
+        attributes.getValue("Implementation-Name"),
+        attributes.getValue("Implementation-Version"),
+        DIALECTS,
+        "https://github.com/networknt/json-schema-validator/",
+        "https://github.com/networknt/json-schema-validator/issues",
+        System.getProperty("os.name"),
+        System.getProperty("os.version"),
+        System.getProperty("java.vendor.version")
+      )
+    );
+    output.println(objectMapper.writeValueAsString(startResponse));
+  }
+
+  private void dialect(JsonNode node) throws JsonProcessingException {
+    DialectRequest dialectRequest = objectMapper.treeToValue(
+      node,
+      DialectRequest.class
+    );
+    if (DIALECTS.contains(dialectRequest.dialect())) {
+      output.println(
+        objectMapper.writeValueAsString(new DialectResponse(true))
+      );
+    } else {
+      output.println(
```

Comment:
I'm not sure if you copied this from @harrel56 's harness (and am too lazy to look at the minute :) but technically this would be a Bowtie bug to ever reach this -- `{"ok": false}` isn't meant to mean "I don't understand that dialect", it's meant to mean "I don't support setting dialects at all outside of a schema" -- Bowtie sending you (the harness) a dialect request that isn't one of the ones supported by the implementation (i.e. that you responded with in `start`) would just be a bug, that should never happen, and would seemingly be the only way one could get here in the harness.

Reply:
So I got it incorrectly, gonna need to adjust it in my harness too

Reply:
(Lemme know if there's some better place to make that even more explicit!)

Reply:
fixed